src/plugins/discover/public/application/doc_views/doc_views_types.ts

```diff
@@ -18,7 +18,9 @@ export interface AngularDirective {
 
 export type AngularScope = IScope;
 
-export type ElasticSearchHit<T = unknown> = estypes.SearchResponse<T>['hits']['hits'][number];
+export type ElasticSearchHit<T = unknown> = estypes.SearchResponse<T>['hits']['hits'][number] & {
+  isAnchor?: boolean;
```

Comment:
`isAnchor` is very specific to the context part of Discover, so it should be omitted to introduce the variable here. Best way would be to get rid of the property (e.g. by using a helper function `isAnchor(esHit)`. However this would need much more work due to legacy code, so it should be solved easier).

 I think `ElasticSearchHit` shouldn't be changed, `EsHitRecord` in `context.ts` should be introduced like `ElasticSearchHit` in this example, und this type should be used in the whole context part of Discover instead of `ElasticSearchHit`